src/john.c

```diff
@@ -1241,19 +1216,22 @@ static void john_load(void)
 	}
 
 	/*
-	 * Nefarious hack and memory leak. Among other problems, we'd want
-	 * ldr_drop_database() after this, but it's built with mem_alloc_tiny()
-	 * so it's not trivial. Works like a champ though, except with
-	 * DEScrypt. I have no idea why, maybe because LM and DES share code?
+	 * Nefarious hack and memory leak. Works like a champ though.
 	 */
-	if (options.flags & FLG_LOOPBACK_CHK &&
-	    database.format != &fmt_LM && database.format != &fmt_DES) {
+	if ((options.flags & FLG_LOOPBACK_CHK) && database.format != &fmt_LM && database.format != &fmt_DES) {
```

Comment:
Doesn't this check unnecessarily prevent having loopback test reassembled LM cracks against descrypt hashes? An obscure case, but not entirely unrealistic.

Reply:
Yes, and that lead to some confusion when I tested this (having one bigcrypt hash and one pwdump entry in a test file, with same passphrase). I'll experiment a little.

Reply:
I now committed a version with no restrictions. I also now restore the `fmt.next` pointers, just in case. Also improved comments.

Reply:
> with no restrictions

BTW I decided it's relevant to activate this even for LM - in really obscure cases, a long password could be permuted by rules and give new ones in ways that the same rules wouldn't end up with for the halves.